bolt-design-system/bolt
# Update and Refresh Drupal Lab Integration Example

## Summary
Maintenance updates and improvements to our Bolt-integrated Drupal Lab instance, now that Bolt v2.0 has been released. Relates to the [BDS-676](http://vjira2:8080/browse/BDS-676) Jira ticket.

These updates include:
- Updating out of date documentation in the README
- Simplifying the install / setup process for non-Bolt developers
- Updating Composer + NPM dependencies
- Upgrading the example theme to use Bolt v2.0
- Renaming the site theme + image assets to the new v2.0 look and feel
- Using the Field Group Drupal module, wire up the Bolt Button component in Drupal Lab as a new Paragraph type available (trying to keep UX and simplicity in mind). Note the collapsible advanced config + helper text with links pointing back to the boltdesignsystem.com site for references!

![edit_page_eligo_eu_ibidem_jus_neque_venio___drupal_lab](https://user-images.githubusercontent.com/1617209/46351944-96735d80-c626-11e8-9d23-592f72aae44c.png)

![edit_page_eligo_eu_ibidem_jus_neque_venio___drupal_lab](https://user-images.githubusercontent.com/1617209/46351959-9ffcc580-c626-11e8-96f3-af54aaa0faaf.png)


## How to test
1. Try running through the install and setup instructions included in the [README](https://github.com/bolt-design-system/bolt/tree/feature/update-drupal-lab-example/example-integrations/drupal-lab) to make sure everything makes sense and works
2. Running Drupal Lab locally, test adding the new Button component paragraph type to confirm the new example is straightforward to use + working as expected

> Note: I originally tried to get the conditional_fields Drupal module in here instead (ie. so certain paragraph fields don't show up or change depending on how the form is interacted with) but unfortunately kept running into PHP errors -- similar to issues other people have reported with integrating Paragraphs with Conditional Fields. 
>
> @remydenton are you familiar with other ways we could get demo conditional field logic as a POC that doesn't require a ton of custom code to get written?
